### ISSUE TITLE
eb does not automatically module Core modules during a robot build

### ISSUE THREAD

**Issue by gwolosh** [easybuilders/easybuild-framework#2186]
```
buildpath            (D) = /afs/cad/u/g/w/xxx/.local/easybuild/build
installpath          (D) = /afs/cad/u/g/w/xxx/.local/easybuild
module-naming-scheme (E) = HierarchicalMNS
repositorypath       (D) = /afs/cad/u/g/w/xxx/.local/easybuild/ebfiles_repo
 robot-paths          (D) = afs/cad/u/g/w/xxx/eb/software/Core/EasyBuild/3.1.2/lib/python2.7/site-packages/easybuild_easyconfigs-3.1.2-py2.7.egg/easybuild/easyconfigs
 sourcepath           (D) = /afs/cad/u/g/w/xxx/.local/easybuild/sources
```

`eb GCC-6.3.0-2.27.eb --robot --installpath=XXXX --installpath-modules =YYYY --robot `
Bison installation failed due to a module load issue.
From the log --

```
 build_log.py:147 ERROR EasyBuild crashed with an error (at ?:124 in __init__): Changing environment as dictated by module failed: name 'false' is
os.environ["MODULEPATH"] = "/afs/cad/u/g/w/XXX/.local/easybuild/modules/all:/etc/modulefiles:/usr/share/modulefiles:/usr/share/modulefiles/Linux:/usr/share/modulefiles/C
false
, stderr: Lmod has detected the following error: The following module(s) are unknown:
"M4/1.4.17"
```

module use YYYY/all/Core and rerun - Builds successfully

**Comment by boegel** [comment]
@geimer Thoughts on this? Should EasyBuild do the `module use .../Core` itself, maybe along with a big fat warning that `$MODULEPATH` isn't set up correctly to consume the modules EasyBuild procudes in the HMNS?

**Comment by geimer** [comment]
@boegel I never tried `eb` with `$MODULEPATH` not being set up correctly.  What does EasyBuild do for the "classical" naming scheme?  Will `modules/all` be added to `$MODULEPATH` automatically?  If so, I think that `Core` should be added when using HMNS.

**Comment by boegel** [comment]
@geimer To get back to this: yes, EasyBuild automatically makes sure that the location where it installs modules too is prepended to `$MODULEPATH.

I agree it makes sense to do the same for an HMNS...

**Comment by Quarkmus** [comment]
@boegel So if I'm using HMNS I have to add those paths manually to the MODULEPATH? 
This confuses me as det_init_modulepaths returns exactly those missing paths, is there no way to automate this?

**Comment by boegel** [comment]
Finally got around to fixing this, see #2857 (to be included in EasyBuild v3.9.1)...

**Comment by boegel** [comment]
The test that was added in #2857 reproduces the reported problem, and confirms that the issue is fixed, so closing this...
